typegate/src/typegate/artifacts/local.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+import { resolve } from "std/path/resolve.ts";
+import { HashTransformStream } from "../../utils/hash.ts";
+import {
+  ArtifactMeta,
+  ArtifactStore,
+  getLocalPath,
+  STORE_DIR,
+  STORE_TEMP_DIR,
+} from "./mod.ts";
+import { createHash } from "node:crypto";
+import * as jwt from "jwt";
+
+export interface UploadUrlStore {
+  mapToMeta: Map<string, ArtifactMeta>;
+  expirationQueue: [string, number][];
+  expirationTimerId: number;
+}
+
+function initUploadUrlStore() {
+  const mapToMeta = new Map<string, ArtifactMeta>();
+  const expirationQueue: [string, number][] = [];
+  const expirationTimerId = setInterval(() => {
```

Comment:
scanning quickly: I am unclear about that feature, I believe it worth explaining this in a short comment on the top of the file

Reply:
Added an overview description of the feature.